internal/provider/resources/deployment_access_test.go

```diff
@@ -0,0 +1,212 @@
+package resources_test
```

Comment:
We don't currently have any tests for block access, which is the closest API to this one, so this may not be anywhere near perfect - but once we tune it we should be able to add some tests for block access according to these.